### PR TITLE
Spinner redesign

### PR DESCRIPTION
**Pull request by libertymayc**
Spinner new sizes and SVGs based on size and density work

### CODE REVIEW

File: packages/lab/src/spinner/svgSpinners/SpinnerDefault.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { SVGAttributes } from "react";
+
+const baseName = "saltSvgSpinner";
```
Comment (joshwooding) [inline packages/lab/src/spinner/svgSpinners/SpinnerDefault.tsx:3]: Is it worth using makePrefixer here?

File: packages/lab/src/spinner/svgSpinners/index.ts [hunk 2]
```diff
@@ -1,14 +1,11 @@
-import { SpinnerSmall } from "./SpinnerSmall";
-import { SpinnerMedium } from "./SpinnerMedium";
+import { SpinnerDefault } from "./SpinnerDefault";
 import { SpinnerLarge } from "./SpinnerLarge";
 
 export const getSvgSpinner = (size: string) => {
```
Comment (joshwooding) [inline packages/lab/src/spinner/svgSpinners/index.ts:4]: Is it worth keeping getSvgSpinner when there are only two choices now?

File: packages/lab/src/spinner/svgSpinners/SpinnerLarge.tsx [hunk 3]
```diff
@@ -2,29 +2,30 @@ import { SVGAttributes } from "react";
 
 const baseName = "saltSvgSpinner";
 
+/* SVG based on MD, do not change viewbox attribute */
 export const SpinnerLarge = (props: SVGAttributes<SVGSVGElement>) => (
-  <svg className={`${baseName}-spinner`} viewBox="0 0 48 48" {...props}>
+  <svg className={`${baseName}-spinner`} viewBox="0 0 56 56" {...props}>
```
Comment (joshwooding) [inline packages/lab/src/spinner/svgSpinners/SpinnerLarge.tsx:7]: Do we need a second SVG? Could we just use the same SVG scaled?

File: packages/lab/src/spinner/Spinner.tsx [hunk 4]
```diff
@@ -64,14 +64,13 @@ export const Spinner = forwardRef<HTMLDivElement, SpinnerProps>(
       disableAnnouncer,
       role = "img",
       className,
-      size = "medium",
+      size = "default",
       id: idProps,
```
Comment (joshwooding) [inline packages/lab/src/spinner/Spinner.tsx:68]:
```suggestion
      id: idProp,
```

File: packages/lab/src/spinner/Spinner.tsx [hunk 5]
```diff
@@ -64,14 +64,13 @@ export const Spinner = forwardRef<HTMLDivElement, SpinnerProps>(
...
       ...rest
     },
     ref
   ) {
     const generatedId = useId(idProps);
```
Comment (joshwooding) [inline packages/lab/src/spinner/Spinner.tsx:73]:
```suggestion
    const id= useId(idProp);
```

File: packages/lab/src/spinner/Spinner.tsx [hunk 6]
```diff
@@ -117,12 +116,12 @@ export const Spinner = forwardRef<HTMLDivElement, SpinnerProps>(
...
       <div
         aria-label={ariaLabel}
-        className={clsx(`${baseName}-${size}`, className, baseName)}
+        className={clsx(withBaseName(), withBaseName(size), className)}
         ref={ref}
         role={role}
         {...rest}
       >
-        <SvgSpinner id={generatedId} />
+        <SpinnerSVG id={generatedId} />
```
Comment (joshwooding) [inline packages/lab/src/spinner/Spinner.tsx:124]:
```suggestion
        <SpinnerSVG id={id} />
```

File: packages/lab/src/spinner/Spinner.tsx (hunk 4 above)
Comment (libertymayc) [reply packages/lab/src/spinner/Spinner.tsx:68]: this was just going to be the new designs and then improv in new pr, but will commit these :)

File: packages/lab/src/spinner/svgSpinners/SpinnerSVG.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import { makePrefixer } from "@salt-ds/core";
+import { SVGAttributes } from "react";
+
+const withBaseName = makePrefixer("saltSvgSpinner");
```
Comment (joshwooding) [inline packages/lab/src/spinner/svgSpinners/SpinnerSVG.tsx:4]: Do you think this should have a different base name?

File: packages/lab/src/spinner/Spinner.tsx [hunk 8]
```diff
@@ -15,9 +15,9 @@ import "./Spinner.css";
 // TODO: documentation -- add line about best practices:
 // - Improve accessibility by customizing the aria-label to provide additional context about *what* is loading, e.g. `aria-label="loading settings panel"`.
 
-export const SpinnerSizeValues = ["small", "medium", "large"] as const;
+export const SpinnerSizeValues = ["default", "large"] as const;
 export type SpinnerSize = typeof SpinnerSizeValues[number];
-const baseName = "saltSvgSpinner";
+const withBaseName = makePrefixer("saltSvgSpinner");
```
Comment (joshwooding) [inline packages/lab/src/spinner/Spinner.tsx:20]: saltSvgSpinner seems a bit weird to me. saltSpinner might be better?

**Review by ImmanuelBaskaran — APPROVED** [review]
Looks good to me

**Pull request merged**